### PR TITLE
Feature/upgrade to latest junix socket dependency

### PR DESCRIPTION
**Pull request by padas2**
Upgrading the version of junixsocket dependencies

---
This change is [Reviewable](https://reviewable.io/reviews/docker-java/docker-java/1159)

### CODE REVIEW

**Comment by dbyron0** [comment]
Is https://github.com/kohlschutter/junixsocket/issues/31 relevant?  Perhaps with version 2.1.0 (or maybe 2.1.2 to get the latest), we no longer need to monkey patch junixsocket here and can delete https://github.com/docker-java/docker-java/blob/master/src/main/java/org/newsclub/net/unix/AFUNIXSocketImpl.java ?  And if we do, I bet we don't need AFUnixSocketException either.

**Comment by padas2** [comment]
> Is [kohlschutter/junixsocket#31](https://github.com/kohlschutter/junixsocket/issues/31) relevant? Perhaps with version 2.1.0 (or maybe 2.1.2 to get the latest), we no longer need to monkey patch junixsocket here and can delete https://github.com/docker-java/docker-java/blob/master/src/main/java/org/newsclub/net/unix/AFUNIXSocketImpl.java ? And if we do, I bet we don't need AFUnixSocketException either.

**Comment by padas2** [comment]
Removed AFUNIXSocketImpl and AFUNIXSocketException classes. Compilation passed. Waiting for travis build results.

**Comment by KostyaSha** [comment]
all socket based tests failed...

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,9 @@
 Change Log
 ===
 
+## 3.1.0-rc-9
```
Comment (KostyaSha) [inline CHANGELOG.md]: 3.1.1 even already released....

Comment (KostyaSha) [reply CHANGELOG.md]: Updated changelog

Comment (padas2) [reply CHANGELOG.md]: Are the socket based test failures because of these changes ?

Comment (KostyaSha) [reply CHANGELOG.md]: no, released version passing unix/tcp tests (beside 2 configurations). This PR fails because amd64 native libs not found something

Comment (padas2) [reply CHANGELOG.md]: I see this error 
Caused by: java.lang.NoClassDefFoundError: Could not initialize class org.newsclub.net.unix.NativeUnixSocket

But I believe this class should be available

**Comment by KostyaSha** [comment]
```
java.lang.UnsatisfiedLinkError: Artifact junixsocket-native-common does not contain library for amd64-Linux	at org.newsclub.net.unix.NativeLibraryLoader.findLibraryArtifact(NativeLibraryLoader.java:70)	at org.newsclub.net.unix.NativeLibraryLoader.loadLibrary(NativeLibraryLoader.java:127)	at org.newsclub.net.unix.NativeUnixSocket.<clinit>(NativeUnixSocket.java:37)	at org.newsclub.net.unix.AFUNIXSocket.<init>(AFUNIXSocket.java:37)	at org.newsclub.net.unix.AFUNIXSocket.newInstance(AFUNIXSocket.java:56)	at com.github.dockerjava.jaxrs.ApacheUnixSocket.<init>(ApacheUnixSocket.java:51)	at com.github.dockerjava.jaxrs.UnixConnectionSocketFactory.createSocket(UnixConnectionSocketFactory.java:66)	at org.apache.http.impl.conn.DefaultHttpClientConnectionOperator.connect(DefaultHttpClientConnectionOperator.java:119)	at org.apache.http.impl.conn.PoolingHttpClientConnectionManager.connect(PoolingHttpClientConnectionManager.java:373)	at org.apache.http.impl.execchain.MainClientExec.establishRoute(MainClientExec.java:394)	at org.apache.http.impl.execchain.MainClientExec.execute(MainClientExec.java:237)	at org.apache.http.impl.execchain.ProtocolExec.execute(ProtocolExec.java:185)	at org.apache.http.impl.execchain.RetryExec.execute(RetryExec.java:89)	at org.apache.http.impl.execchain.RedirectExec.execute(RedirectExec.java:110)	at org.apache.http.impl.client.InternalHttpClient.doExecute(InternalHttpClient.java:185)	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:72)	at org.glassfish.jersey.apache.connector.ApacheConnector.apply(ApacheConnector.java:450)	at org.glassfish.jersey.client.ClientRuntime.invoke(ClientRuntime.java:278)	at org.glassfish.jersey.client.JerseyInvocation.lambda$invoke$1(JerseyInvocation.java:767)	at org.glassfish.jersey.internal.Errors.process(Errors.java:316)	at org.glassfish.jersey.internal.Errors.process(Errors.java:298)	at org.glassfish.jersey.internal.Errors.process(Errors.java:229)	at org.glassfish.jersey.process.internal.RequestScope.runInScope(RequestScope.java:414)	at org.glassfish.jersey.client.JerseyInvocation.invoke(JerseyInvocation.java:765)	at org.glassfish.jersey.client.JerseyInvocation$Builder.method(JerseyInvocation.java:428)	at org.glassfish.jersey.client.JerseyInvocation$Builder.get(JerseyInvocation.java:324)	at com.github.dockerjava.jaxrs.InspectImageCmdExec.execute(InspectImageCmdExec.java:27)	at com.github.dockerjava.jaxrs.InspectImageCmdExec.execute(InspectImageCmdExec.java:13)	at com.github.dockerjava.jaxrs.AbstrSyncDockerCmdExec.exec(AbstrSyncDockerCmdExec.java:23)	at com.github.dockerjava.core.command.AbstrDockerCmd.exec(AbstrDockerCmd.java:35)	at com.github.dockerjava.core.command.InspectImageCmdImpl.exec(InspectImageCmdImpl.java:40)	at com.github.dockerjava.junit.DockerRule.before(DockerRule.java:80)	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:46)	at org.junit.rules.RunRules.evaluate(RunRules.java:20)	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)	at org.apache.maven.surefire.junitcore.pc.Scheduler$1.run(Scheduler.java:393)	at org.apache.maven.surefire.junitcore.pc.InvokerStrategy.schedule(InvokerStrategy.java:54)	at org.apache.maven.surefire.junitcore.pc.Scheduler.schedule(Scheduler.java:352)	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)	at org.junit.runners.Suite.runChild(Suite.java:128)	at org.junit.runners.Suite.runChild(Suite.java:27)	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)	at org.apache.maven.surefire.junitcore.pc.Scheduler$1.run(Scheduler.java:393)	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)	at java.util.concurrent.FutureTask.run(FutureTask.java:266)	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)	at jav
```

File: pom.xml [hunk 2]
```diff
@@ -67,7 +67,7 @@
 		<slf4j-api.version>1.7.25</slf4j-api.version>
 
 		<bouncycastle.version>1.60</bouncycastle.version>
-		<junixsocket.version>2.0.4</junixsocket.version>
+		<junixsocket.version>2.1.0</junixsocket.version>
```
Comment (dbyron0) [inline pom.xml]:
```suggestion
		<junixsocket.version>2.1.2</junixsocket.version>
```
Does this by chance make the failure go away?

Comment (KostyaSha) [reply pom.xml]: let's try

File: pom.xml [hunk 3]
```diff
@@ -67,7 +67,7 @@
 		<slf4j-api.version>1.7.25</slf4j-api.version>
 
 		<bouncycastle.version>1.60</bouncycastle.version>
-		<junixsocket.version>2.0.4</junixsocket.version>
+		<junixsocket.version>2.1.2</junixsocket.version>
```
Comment (dbyron0) [inline pom.xml]:
```suggestion
		<junixsocket.version>2.2.0</junixsocket.version>
```

Comment (dbyron0) [reply pom.xml]: Let's see if this gets us farther.  Also see https://github.com/kohlschutter/junixsocket/issues/31#issuecomment-464527380 for more suggestions.

Comment (kohlschuetter) [reply pom.xml]: Yes, please directly update to 2.2.0. There's no point in moving anymore.

File: pom.xml (hunk 2 above)
Comment (kohlschuetter) [reply pom.xml]: Please update to 2.2.0 (the latest version)

**Pull request merged**